samza-core/src/main/scala/org/apache/samza/system/chooser/DefaultChooser.scala

```diff
@@ -251,7 +255,7 @@ class DefaultChooser(
    * Defines a mapping from SystemStream name to SystemAdmin.
    * This is useful for determining if a bootstrap SystemStream is caught up.
    */
-  systemAdmins: Map[String, SystemAdmin] = Map()) extends MessageChooser with Logging {
+  systemAdmins: SystemAdmins = new SystemAdmins()) extends MessageChooser with Logging {
```

Comment:
Are these empty default SystemAdmins there for tests? If so, I'd prefer to just make them required and explicitly pass on in tests.

Reply:
Yes this default value is only used for tests. If we were to make them required, should we also make other parameters required as well? I can make this change if you prefer.

Reply:
If it's not too much work, making this param alone required would be fine. I mostly want to get rid of the SystemAdmins default constructor.